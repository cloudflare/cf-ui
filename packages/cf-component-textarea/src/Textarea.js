const React = require('react');
const {PropTypes} = React;
const requiredIf = require('react-required-if');

class Textarea extends React.Component {

  render() {
    const {className, ...props} = this.props;
    const classes = ['cf-textarea', className];

    if (this.props.disabled) classes.push('cf-textarea--disabled');
    if (this.props.readOnly) classes.push('cf-textarea--readonly');

    return <textarea className={classes.join(' ')} {...props} />;
  }
  
}

Textarea.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  readOnly: PropTypes.bool,
  onChange: requiredIf(PropTypes.func, props => !props.disabled && !props.readOnly),
  onFocus: PropTypes.func,
  onBlur: PropTypes.func
};

module.exports = Textarea;
