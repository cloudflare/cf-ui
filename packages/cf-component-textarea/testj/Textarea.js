import React from 'react';
import renderer from 'react-test-renderer';
import Textarea from 'cf-component-textarea';

describe('Textarea', () => {
  test('should render', () => {
    const component = renderer.create(
      <Textarea name="example" value="content" onChange={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
