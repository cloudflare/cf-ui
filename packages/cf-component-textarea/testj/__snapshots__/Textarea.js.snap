// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea should render 1`] = `
<textarea
  className="cf-textarea"
  name="example"
  onChange={[Function]}
  value="content"
/>
`;
